src/XMakeTasks/UnitTests/Exec_Tests.cs

```diff
@@ -85,7 +85,9 @@ public void Timeout()
             Assert.Equal(-1, exec.ExitCode);
             ((MockEngine)exec.BuildEngine).AssertLogContains("MSB5002");
             Assert.Equal(1, ((MockEngine)exec.BuildEngine).Warnings);
-            Assert.Equal(1, ((MockEngine)exec.BuildEngine).Errors);
+
+            // ToolTask does not log an error on timeout.
+            Assert.Equal(0, ((MockEngine)exec.BuildEngine).Errors);
```

Comment:
Agreed, this one scared me a bit that it's just a warning. I even checked the code and stepped through to be sure. It appears intentional, but I didn't get a chance to see if that changed at some point.